features/251-templates/index.md

```diff
@@ -0,0 +1,177 @@
+# Templates
+
+## Overview
+
+Skygear has many features to send message where message is either email or SMS.
+Some features like Forgot Password allows the developer to customize the HTML page.
+This document introduces a new way to organize templates with localization support.
+
+
+## Template
+
+The template is represented by the following struct
```

Comment:
What would happen to existing template configs in `UserConfiguration`?

Reply:
For templates that are email or SMS, the configs are all removed.
For templates that are HTML web page, only the external URL config is kept, other configs are removed.

Reply:
For external URL config, does it means the template is hosted at the URL, and auth gear would fetch and render the template? If so, it could be a security concern, since the URL may set to be an internal service accessible only to auth gear.

Reply:
No. It is 302 redirect.

Reply:
So for example `ResetSuccessHTMLURL` of `ForgotPasswordConfiguration`, the web-page would not be able to render relevant information, such as user email? Or do we pass the existing template context to it as query params?

Reply:
The relevant code is in `pkg/auth/dependency/forgotpwdemail/resetpwd_html.go`. You're right. The context is included in the URL query. We can further review what to include in the context though it is unrelated to the proposed changes in templates.